ui/app/components/secret-engine/configure-aws.ts

```diff
@@ -5,78 +5,130 @@
-    @accessKey={{accessKey}}
-    @secretKey={{secretKey}}
-    @region={{region}}
-    @iamEndpoint={{iamEndpoint}}
-    @stsEndpoint={{stsEndpoint}}
-    @saveAWSRoot={{action "save" "saveAWSRoot"}}
-    @saveAWSLease={{action "save" "saveAWSLease"}} />
+    @rootConfig={{this.model.aws-root-config}}
+    @leaseConfig={{this.model.aws-lease-config}} 
+    @id={{this.model.id}} 
+    />
  * ```
  *
- * @param {object} model - aws secret engine model
- * @param {string} tab - current tab selection
- * @param {string} accessKey - AWS access key
- * @param {string} secretKey - AWS secret key
- * @param {string} region - AWS region
- * @param {string} iamEndpoint - IAM endpoint
- * @param {string} stsEndpoint - Sts endpoint
- * @param {Function} saveAWSRoot - parent action which saves AWS root credentials
- * @param {Function} saveAWSLease - parent action which updates AWS lease information
- *
+ * @param {object} rootConfig - AWS config/root model
+ * @param {object} leaseConfig - AWS config/lease model
+ * @param {string} id - name of the AWS secret engine, ex: 'aws-123'
  */
 
-type AWSRootCredsFields = {
-  access_key: string | null;
-  iam_endpoint: string | null;
-  sts_endpoint: string | null;
-  secret_key: string | null;
-  region: string | null;
-};
-
-type LeaseFields = { lease: string; lease_max: string };
-
 interface Args {
-  model: SecretEngineModel;
-  tab?: string;
-  accessKey: string;
-  secretKey: string;
-  region: string;
-  iamEndpoint: string;
-  stsEndpoint: string;
-  saveAWSRoot: (data: AWSRootCredsFields) => void;
-  saveAWSLease: (data: LeaseFields) => void;
+  leaseConfig: LeaseConfigModel;
+  rootConfig: RootConfigModel;
+  id: string;
 }
 
 export default class ConfigureAwsComponent extends Component<Args> {
-  @action
-  saveRootCreds(data: AWSRootCredsFields, event: Event) {
+  @service declare readonly router: Router;
+  @service declare readonly store: Store;
+  @service declare readonly flashMessages: FlashMessageService;
+
+  @tracked errorMessageRoot: string | null = null;
+  @tracked errorMessageLease: string | null = null;
+  @tracked invalidFormAlert: string | null = null;
+  @tracked modelValidationsLease: ValidationMap | null = null;
+
+  @task
+  @waitFor
+  *save(event: Event) {
     event.preventDefault();
-    this.args.saveAWSRoot(data);
+    this.resetErrors();
+    const { id, leaseConfig, rootConfig } = this.args;
+    // Note: aws/root-config model does not have any validations
+    const isValid = this.validate(leaseConfig);
+    if (!isValid) return;
+    // Check if any of the models' attributes have changed.
+    // If no changes to either model, transition and notify user.
+    // If changes to one model, save the model that changed and notify user.
+    // Otherwise save both models.
+    // Note: "backend" dirties model state so explicity ignore it here.
+    const leaseAttrChanged =
+      Object.keys(leaseConfig.changedAttributes()).filter((item) => item !== 'backend').length > 0;
```

Comment:
Thankfully the AWS models all have simple attribute types (string, number, etc) but just a flag here that if we add attributes to the model that are complex (arrays, objects) then `changedAttributes` will not reflect changes to those values. This may have been fixed since I ran into this issue, but something to be aware of for future improvements